### PR TITLE
feat(pipeline): notify failure on slack

### PR DESCRIPTION
**Pull request by nija-at**
The pipeline now optionally accepts a list of slack channels
registered with AWS Chatbot. These channels receive notification upon
any failure in the pipeline.

-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by RomainMuller — APPROVED** [review]

File: lib/pipeline.ts [hunk 1]
```diff
@@ -160,7 +161,13 @@ export interface PipelineProps {
...
    * @default - Duration.hours(8)
    */
-  buildTimeout?: Duration;
+  readonly buildTimeout?: Duration;
+
+  /**
+   * Slack channel configurations where failure notification
+   * in this pipeline should be sent.
+   */
+  readonly failureNotifySlack?: chatbot.SlackChannelConfiguration[];
```
Comment (rix0rrr) [inline lib/pipeline.ts:170]: Would have preferred for failure notifications to have been an enum-like type, to pick between Slack and Chime .

File: lib/pipeline.ts [hunk 2]
```diff
@@ -286,6 +293,23 @@ export class Pipeline extends Construct {
     if (props.autoBuild) {
       this.autoBuild(props.autoBuildOptions);
     }
+
+    if (props.failureNotifySlack && props.failureNotifySlack.length > 0) {
+      props.failureNotifySlack.forEach(s => {
+        new starnotifs.CfnNotificationRule(this, `FailureSlackNotification-${s.slackChannelConfigurationName}`, {
```
Comment (rix0rrr) [inline lib/pipeline.ts:299]: Thanks CodeStar! Good find!

File: lib/pipeline.ts [hunk 3]
```diff
@@ -286,6 +293,23 @@ export class Pipeline extends Construct {
...
+          name: `${this.pipeline.pipelineName}-failednotifications`,
+          detailType: 'BASIC',
```
Comment (rix0rrr) [inline lib/pipeline.ts:301]: Now I'm curious about more options here.

File: lib/pipeline.ts [hunk 4]
```diff
@@ -286,6 +293,23 @@ export class Pipeline extends Construct {
...
+          resource: this.pipeline.pipelineArn,
+          targets: [
+            {
```
Comment (rix0rrr) [inline lib/pipeline.ts:304]: Wouldn't it have made more sense to put the array here instead of `forEach`ing over it? 

Or is this a fake array?

**Pull request merged**